src/coreclr/nativeaot/Runtime/eventpipe/ep-rt-aot.cpp

```diff
@@ -297,10 +303,9 @@ ep_rt_aot_current_thread_get_id (void)
     STATIC_CONTRACT_NOTHROW;
 
 #ifdef TARGET_UNIX
-    // shipping criteria: no EVENTPIPE-NATIVEAOT-TODO left in the codebase
-    // TODO: AOT doesn't have PAL_GetCurrentOSThreadId, as CoreCLR does.
-    // PalDebugBreak();    
-    return static_cast<ep_rt_thread_id_t>(0);
+    // AOT doesn't have PAL_GetCurrentOSThreadId, as CoreCLR does.
+    // But PalGetCurrentThreadIdForLogging does the same thing
```

Comment:
Should we rename it to PalGetCurrentThreadId or PalGetCurrentOSThreadId if that is what its purpose seems to be now?

Reply:
FWIW, the original name was `GetCurrentThreadId`. It was renamed to `PalGetCurrentThreadIdForLogging` in https://github.com/dotnet/corert/commit/9a04f2c1ccf5f7ce664e1ad21998d125f57901aa and that commit has the reasoning.

Reply:
Mono implementation of this method just returns `pthread_self` that is the opaque thread ID and that is very cheap to get (it just returns thread local static):

https://github.com/dotnet/runtime/blob/2b122d36e02355850aef3b9e2bf670ea259c801c/src/mono/mono/eventpipe/ep-rt-mono.h#L962
https://github.com/dotnet/runtime/blob/2b122d36e02355850aef3b9e2bf670ea259c801c/src/mono/mono/utils/mono-threads-posix.c#L242

Similarly, the CoreCLR implementation built on top of Win32 PAL emulator is fairly cheap.

`PalGetCurrentThreadIdForLogging` is implemented as a raw `syscall(SYS_gettid)` that will be at least 100x more expensive than `pthread_self`.

This smells like a pretty significant performance issue.

Reply:
Opened https://github.com/dotnet/runtime/issues/86828 so that this does not get forgotten.

Reply:
> Opened #86828 so that this does not get forgotten.

Thank you! The name did raise red flags (which is why I looked up history) but I didn't realize there would be perf implications. Now the name makes even more sense. When renaming it back, we should make sure there's a comment on that.